### PR TITLE
split asio_util.hpp

### PR DESCRIPTION
**Pull request by PikachuHyA**
## Why

examples like `block_echo_server` do not depend on `async_simple`

before

```mermaid
flowchart TD
   id2(asio_util.hpp)
   id2 --> asio
   id2 --> async_simple
```
after split `asio_util.hpp` into `asio_util.hpp` and `asio_coro_util.hpp`

```mermaid
flowchart TD
   id1(asio_util.hpp)
   id2(asio_coro_util.hpp)
   id1 --> asio
   id2 --> asio
   id2 --> async_simple
```

### CODE REVIEW

File: demo_example/asio_coro_util.hpp [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    std::string port_;
+    std::error_code ec_{};
+};
+
+inline async_simple::coro::Lazy<std::error_code> async_connect(
+    asio::io_context &io_context, asio::ip::tcp::socket &socket,
+    const std::string &host, const std::string &port) noexcept {
+    co_return co_await ConnectAwaiter{io_context, socket, host, port};
+}
+#endif
```
Comment (ChuanqiXu9) [inline demo_example/asio_coro_util.hpp:236]: Add the newline.

**Review by ChuanqiXu9 — COMMENTED** [review]
Why do we need the change?

**Comment by PikachuHyA** [comment]
> Why do we need the change?

when I adjust CMakeLists.txt, the dependences of some examples like `block_echo_server` is not clear.
If `async_simple` is unnecessary for `block_echo_server`, we should not link the target `async_simple`. Thus, files like `async_simple/coro/Lazy.h` is invisible. However, the original `asio_util.hpp` uses that files.

File: demo_example/asio_util.hpp [hunk 2]
```diff
@@ -15,15 +15,7 @@
...
-
-#include <async_simple/coro/Lazy.h>
-#include <async_simple/coro/SyncAwait.h>
-#include <async_simple/executors/SimpleExecutor.h>
-#include <cstdlib>
-#include <iostream>
-#include <thread>
-#include "asio.hpp"
-
+#include <asio.hpp>
```
Comment (ChuanqiXu9) [inline demo_example/asio_util.hpp:18]: Originally, we includes "asio.hpp"

File: demo_example/asio_coro_util.hpp [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef ASYNC_SIMPLE_DEMO_ASIO_CORO_UTIL_H
+#define ASYNC_SIMPLE_DEMO_ASIO_CORO_UTIL_H
+#include <async_simple/coro/Lazy.h>
+#include <async_simple/coro/SyncAwait.h>
+#include <async_simple/executors/SimpleExecutor.h>
+
+#include <asio.hpp>
```
Comment (ChuanqiXu9) [inline demo_example/asio_coro_util.hpp:22]: Use "asio.hpp"

**Review by ChuanqiXu9 — APPROVED** [review]
Makes sense.

**Review by ChuanqiXu9 — CHANGES_REQUESTED** [review]
Approve mistaken.

**Review by ChuanqiXu9 — APPROVED** [review]
LGTM

**Pull request merged**